src/protocol.rs

```diff
@@ -28,6 +32,67 @@ const DISCRIMINANT_LEN: usize = 1;
 
 type Result<T, E = Error> = std::result::Result<T, E>;
 
+pub async fn spawn(port: u16) -> std::io::Result<Vec<tokio::task::JoinHandle<()>>> {
+    let ipv4: SocketAddr = (std::net::Ipv4Addr::UNSPECIFIED, port).into();
+    let ipv6: SocketAddr = (std::net::Ipv6Addr::UNSPECIFIED, port).into();
+    let mut tasks = Vec::new();
+
+    tracing::info!(%port, "spawning IPv4 and IPv6 QCMP sockets");
+    for (address, socket) in [
+        (ipv4, tokio::net::UdpSocket::bind(ipv4).await?),
```

Comment:
Looks like this is the issue - attempting to bind both an ipv4 address and an ipv6 address to the same port is causing it to already have the same port in use.

If you want to do this, I expect you'll want to use / extend `utils::net::socket_with_reuse` so that you can reuse the port. You will want to listen to both sockets, but that should be handle-able with a `select!`